x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/schema_logic.test.ts

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogicMounter, mockFlashMessageHelpers, mockHttpValues } from '../../../__mocks__';
+import { mockEngineActions } from '../../__mocks__/engine_logic.mock';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { SchemaType, Schema } from '../../../shared/schema/types';
+
+import { SchemaLogic } from './schema_logic';
+
+describe('SchemaLogic', () => {
+  const { mount } = new LogicMounter(SchemaLogic);
+  const { http } = mockHttpValues;
+  const { flashAPIErrors, flashSuccessToast, setErrorMessage } = mockFlashMessageHelpers;
+
+  const MOCK_RESPONSE = {
+    schema: {
+      some_text_field: SchemaType.Text,
+      some_number_field: SchemaType.Number,
+    },
+    mostRecentIndexJob: {
+      percentageComplete: 100,
+      numDocumentsWithErrors: 10,
+      activeReindexJobId: 'some-id',
+      isActive: false,
+      hasErrors: true,
+    },
```

Comment:
FYI that a large majority of this data we're no longer using after 7.12/the document storage refactor - `percentageComplete` will always come back as 100 and `isActive` as false. I still kept it in here as an example, but def would be nice to clean up our APIs some day